src/player.cpp

```diff
@@ -12251,26 +12251,36 @@ int player::encumb( body_part bp ) const
 int player::encumb(body_part bp, double &layers, int &armorenc) const
 {
     int ret = 0;
-    int layer[MAX_CLOTHING_LAYER] = { };
+    // First is the total of encumbrance on the given layer,
+    // and second is the highest encumbrance of any one item on the layer.
+    std::array<std::pair<int, int>, MAX_CLOTHING_LAYER> layer = {{{0,0},{0,0},{0,0},{0,0},{0,0}}};
```

Comment:
```std::array<std::pair<int, int>, MAX_CLOTHING_LAYER> layer();```
Did not do what I wanted, leaving it with explicit initialization for the moment.
For the spurious warning, can we do ```-Wno-missing-field-initializers``` just for compiler versions that complain?
ISTR it was some of the mingw versions and older GCCs, but might be misremembering.  But as long as some of the auto-builds still check properly (e.g. clang and newer GCCs) we can leave it on there and get coverage.